src/storage/table/row_group.cpp

```diff
@@ -318,6 +325,19 @@ void RowGroup::TemplatedScan(Transaction *transaction, RowGroupScanState &state,
 			auto lowest_active_start = transaction_manager.LowestActiveStart();
 			auto lowest_active_id = transaction_manager.LowestActiveId();
 
+			count = state.row_group->GetCommittedSelVector(lowest_active_start, lowest_active_id, state.vector_index,
+			                                               valid_sel, max_count);
+			if (count == 0) {
+				// nothing to scan for this vector, skip the entire vector
+				NextVector(state);
+				continue;
+			}
+		} else if (TYPE == TableScanType::TABLE_SCAN_COMMITTED_ROWS_OMIT_PERMANENTLY_DELETED_CHECKPOINT) {
+			auto &transaction_manager = TransactionManager::Get(db);
+			auto lowest_active_start = transaction_manager.LowestActiveStart();
+			// We have to decrement the lowest active transaction id by one to ensure we scan all the tuples correctly
+			auto lowest_active_id = transaction_manager.LowestActiveId() - 1;
```

Comment:
Could you perhaps expand on why this is necessary?

Reply:
During debugging we found that, since we're scanning the row group just before checkpointing, that the transaction id has already been incremented at this point. Meaning if we don't do this, we miss out on tuples that were deleted in the previous transactions (and are in the VersionInfo) and end up writing these to disk.